### PR TITLE
Sano/cf 96/fill up sidebar and refactored redux store

### PR DESCRIPTION
**Pull request by Snuzzn**
- Course menu content is hooked up to the api
image
- Unused redux actions/reducers were removed
- Refactored CourseMenu.jsx
- Renamed degree selector to degree wizard to avoid repetition/confusion between 'degree selector' and 'course selector'.

### CODE REVIEW

**Comment by Peedee2002** [comment]
do you have a screenshot of the stuff being hooked up?

File: frontend/src/pages/CourseSelector/courseDescription/CourseDescription.jsx [hunk 1]
```diff
@@ -85,115 +83,126 @@ export default function CourseDescription() {
...
+                  showIcon
+                  closable
+                  afterClose={() => setShow(false)}
+                />
+              )}
+              {show && !courseInPlanner && (
+                <Alert
+                  message={`Successfully removed ${id} from your planner!`}
+                  type="success"
+                  style={{ marginBottom: "1rem" }}
```
Comment (Peedee2002) [inline frontend/src/pages/CourseSelector/courseDescription/CourseDescription.jsx]: can this styling be pushed to a less file?

File: frontend/src/pages/TermPlanner/DragDropLogic.js [hunk 2]
```diff
@@ -13,18 +12,17 @@ export const handleOnDragEnd = (result, dragEndProps) => {
   if (!destination) return; // drag outside container
 
   // check if prereqs are complete, and trigger warning if not
-  let arePrereqsCompleted = checkPrereq(
-    draggableId,
-    destination.droppableId,
-    courses
-  );
+  // let arePrereqsCompleted = checkPrereq(
+  //   draggableId,
```
Comment (Peedee2002) [inline frontend/src/pages/TermPlanner/DragDropLogic.js]: why not delete?

File: frontend/src/pages/TermPlanner/DragDropLogic.js [hunk 3]
```diff
@@ -104,37 +91,38 @@ export const handleOnDragStart = (
...
+//       if (courses.get(elem) == null) {
+//         // prereq not in term planner
+//         isComplete.set(elem, false);
+//       } else if (courses.get(elem)["plannedFor"] == null) {
+//         // prereq in unplanned
+//         isComplete.set(elem, false);
+//       } else if (term <= courses.get(elem)["plannedFor"]) {
+//         // course placed before (or during) prereq is complete
+//         isComplete.set(elem, false);
+//       } else {
```
Comment (Peedee2002) [inline frontend/src/pages/TermPlanner/DragDropLogic.js]: why not delete?

File: frontend/src/pages/TermPlanner/HideYearTooltip.jsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+      type: "error",
+      message: "Something's not right",
+      description: "You cannot hide all years in your term planner",
+      duration: 2,
+    });
+  };
+
+  const hideYear = (year) => {
+    let i = 0;
+    for (const [key, value] of Object.entries(hidden)) if (value) i++;
```
Comment (Peedee2002) [inline frontend/src/pages/TermPlanner/HideYearTooltip.jsx]: there is nothing wrong with this line, it just facinates me lol

File: frontend/src/reducers/plannerReducer.js [hunk 5]
```diff
@@ -361,3 +354,46 @@ const plannerReducer = (state = initialState, action) => {
 };
 
 export default plannerReducer;
+
+// dummyMap.set("COMP2521", {
+//   title: "Data Structures and Algorithms",
+//   type: "Core",
+//   termsOffered: ["T0", "T1", "T2", "T3"],
+//   prereqs: "COMP1511 && (COMP1521 || DEFAULT3000)",
+//   plannedFor: null,
```
Comment (Peedee2002) [inline frontend/src/reducers/plannerReducer.js:363]: why not delete?

**Review by Peedee2002 — COMMENTED** [review]
this is a massive PR lol - i guess we will get a few of those as we get to MVP as you hook stuff up to backend.

File: frontend/src/pages/TermPlanner/HideYearTooltip.jsx (hunk 4 above)
Comment (Snuzzn) [reply frontend/src/pages/TermPlanner/HideYearTooltip.jsx]: I've now refined this a bit using Object.values()

File: frontend/src/reducers/plannerReducer.js (hunk 5 above)
Comment (Snuzzn) [reply frontend/src/reducers/plannerReducer.js:363]: I've now deleted all but one. I kept one just so its clear what the structure of the course is. Might be useful for reference when working on the reducer.

File: frontend/src/pages/CourseSelector/courseDescription/CourseDescription.jsx (hunk 1 above)
Comment (Snuzzn) [reply frontend/src/pages/CourseSelector/courseDescription/CourseDescription.jsx]: We encountered an issue where certain ant design components overrode the less styles that we gave. I've now solved this by applying an !important to less style. It's not ideal to use this but I think in this case, we can get away with it.

Comment (Peedee2002) [reply frontend/src/pages/CourseSelector/courseDescription/CourseDescription.jsx]: Yea front end libraries are usually a disaster with this type of thing lol - lgtm

File: frontend/src/reducers/plannerReducer.js (hunk 5 above)
Comment (Peedee2002) [reply frontend/src/reducers/plannerReducer.js:363]: Oh my bad I didn't notice that this was documenting how to use the the function - that's fine.

File: frontend/src/pages/CourseSelector/courseDescription/CourseDescription.jsx [hunk 6]
```diff
@@ -85,115 +83,126 @@ export default function CourseDescription() {
...
-            )}
-            <div className="cs-desc-title-bar">
-              <Title level={2} className="text">
-                {id} - {course.title}
+    <div>
+      <div className="cs-description-root">
+        {!pageLoaded ? (
+          <Loading />
+        ) : (
+          <>
```
Comment (Peedee2002) [inline frontend/src/pages/CourseSelector/courseDescription/CourseDescription.jsx:91]: should this be a div?

**Review by Peedee2002 — APPROVED** [review]
would like to see some screenshots, but ill preemptively approve anyways

**Comment by Snuzzn** [comment]
@Peedee2002, pretty much all of this stuff (except course selector menu) is structural changes to make the codebase cleaner. The only real visual change would be the sidebar (I added that screenshot to the description).

**Pull request merged**